src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,257 @@
+                boolean viewCreated = false;
+
+                // Create view only if new or reuse of provided view is not possible
+                if (convertView == null
+                        || !(convertView.findViewById(R.id.wifiName) instanceof CheckBox)) {
+                    LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+                    convertView = inflater.inflate(R.layout.preference_wifi_ssid_row, parent, false);
+                    viewCreated = true;
+                }
+
+                // SSIDs are formatted "name" if they represent an UTF-8 string,
+                // or as 01234567890ABCDEF if hex, but displaying the '"' doesn't look good,
+                // so strip them
+                String displayName = wifi.SSID;
+                if (displayName != null && displayName.length() > 0 && displayName.charAt(0) == '"') {
+                    // If th SSID starts with '"', it is a string name, otherwise a hex string
+                    if (displayName.length() >= 2) {
+                        // actually, this should always be true, otherwise it's a violation of the
+                        // contract of how SSID names are formatted by WifiManager
+                        displayName = displayName.substring(1, displayName.length() - 1);
+                    }
+                }
+
+                CheckBox wifiView = (CheckBox) convertView.findViewById(R.id.wifiName);
+                wifiView.setTag(wifi.SSID); // the SSID this possibly recylced view is currently representing6
+                wifiView.setText(displayName);
+                wifiView.setChecked(mSelectedSsids.contains(wifi.SSID));
+
+                if (viewCreated) {
+                    wifiView.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                        @Override
+                        public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                            // since a view may be reused, it's not possible to capture the value
+                            // of the SSID with a final variable, but it's necessary to transport
+                            // it via the CompundButton's tag
+                            String ssid = (String) buttonView.getTag();
+                            if (isChecked) {
+                                mSelectedSsids.add(ssid);
+                            } else {
+                                mSelectedSsids.remove(ssid);
+                            } // if/else
+                            updateSaveButtonState();
+                        }
+                    });
+                }
+                return convertView;
+            } else {
+                // WiFi is most likely turned off
+                LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
```

Comment:
Actually it's a fallback for a very unlikely race-condition where the WiFi service becomes unavailable between the check in `SettingsFragment` and actually building the list here. Rather unlikely, but since I already had the code from the previous iterations, I left it in as a fallback.